admin_setting.go

```diff
@@ -0,0 +1,461 @@
+// Note that admin settings are only available in Terraform Enterprise.
+//
+// TFE API docs: https://www.terraform.io/docs/cloud/api/admin/settings.html
+type AdminSettings interface {
+	// GetGeneral returns the general settings
+	GetGeneral(ctx context.Context) (*AdminGeneralSetting, error)
+
+	// UpdateGeneral updates general settings.
+	UpdateGeneral(ctx context.Context, options AdminGeneralSettingsUpdateOptions) (*AdminGeneralSetting, error)
+
+	// GetCostEstimation returns the cost estimation settings.
+	GetCostEstimation(ctx context.Context) (*AdminCostEstimationSetting, error)
+
+	// UpdateCostEstimation updates the cost estimation settings.
+	UpdateCostEstimation(ctx context.Context, options AdminCostEstimationSettingOptions) (*AdminCostEstimationSetting, error)
+
+	// GetSAML returns the SAML settings.
+	GetSAML(ctx context.Context) (*AdminSAMLSetting, error)
+
+	// UpdateSAML updates the SAML settings.
+	UpdateSAML(ctx context.Context, options AdminSAMLSettingsUpdateOptions) (*AdminSAMLSetting, error)
+
+	// RevokeSAMLIdpCert revokes the older IdP certificate when the new IdP
+	// certificate is known to be functioning correctly.
+	RevokeSAMLIdpCert(ctx context.Context) (*AdminSAMLSetting, error)
+
+	// GetSMTP returns the SMTP settings.
+	GetSMTP(ctx context.Context) (*AdminSMTPSetting, error)
+
+	// UpdateSMTP updates SMTP settings.
+	UpdateSMTP(ctx context.Context, options AdminSMTPSettingsUpdateOptions) (*AdminSMTPSetting, error)
+
+	// GetTwilio returns the Twilio settings.
+	GetTwilio(ctx context.Context) (*AdminTwilioSetting, error)
+
+	// UpdateTwilio updates Twilio settings.
+	UpdateTwilio(ctx context.Context, options AdminTwilioSettingsUpdateOptions) (*AdminTwilioSetting, error)
+
+	// VerifyTwilio verifies Twilio settings.
+	VerifyTwilio(ctx context.Context, options AdminTwilioSettingsVerifyOptions) error
+
+	// GetCustomization returns the customization settings.
+	GetCustomization(ctx context.Context) (*AdminCustomizationSetting, error)
+
+	// UpdateCustomization updates the customization settings.
+	UpdateCustomization(ctx context.Context, options AdminCustomizationSettingsUpdateOptions) (*AdminCustomizationSetting, error)
+}
+
+// adminSettings implements AdminSettings.
+type adminSettings struct {
+	client *Client
+}
+
+// AdminGeneralSetting represents a the general settings in Terraform Enterprise.
+type AdminGeneralSetting struct {
+	ID                            string `jsonapi:"primary,general-settings"`
+	LimitUserOrganizationCreation bool   `jsonapi:"attr,limit-user-organization-creation"`
+	APIRateLimitingEnabled        bool   `jsonapi:"attr,api-rate-limiting-enabled"`
+	APIRateLimit                  int    `jsonapi:"attr,api-rate-limit"`
+	SendPassingStatusesEnabled    bool   `jsonapi:"attr,send-passing-statuses-for-untriggered-speculative-plans"`
+	AllowSpeculativePlansOnPR     bool   `jsonapi:"attr,allow-speculative-plans-on-pull-requests-from-forks"`
```

Comment:
Seems like there's quite a few settings missing from this list; the documentation is well out of date 😕 You'll want to double check what's actually there and compare that to here.

Reply:
Will do

Reply:
Added